src/App/Version/TH.hs

```diff
@@ -0,0 +1,90 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module App.Version.TH
+  ( getCurrentTag,
+  )
+where
+
+import Control.Carrier.Diagnostics (resultValue, runDiagnostics)
+import Control.Effect.Diagnostics
+  ( Diagnostics,
+    fromEitherShow,
+  )
+import qualified Data.ByteString.Lazy as BSL
+import Data.Maybe (fromMaybe)
+import Data.Text (Text)
+import qualified Data.Text.Encoding as TE
+import qualified Data.Text as T
+import Data.Versions
+import Effect.Exec
+import GitHash
+import Instances.TH.Lift ()
+import Language.Haskell.TH
+import Path
+import Language.Haskell.TH.Syntax (addDependentFile)
+import Path.IO (getCurrentDir)
+
+
+gitTagPointCommand :: Text -> Command
```

Comment:
I wouldn't call this a function in the normal sense.  It's more like a parametric constant.  The only way to build the full `Command` type is to create the command all at once.  But almost all of it stays the same all of the time.  If there was a way to set up a method of applying the `commit` argument later, then this would actually be a constant.  And constants generally live at the top.